Add unformatted paste button

erilyth opened the pull request:
Links and formatted text will be pasted using paste_special from AbiWord. However for images it would do the same thing as a normal paste would do.
Currently the widget on the toolbar is the same as the paste button.
(If this is fine I will also add another icon for paste unformatted and link this button to that and make a PR)

samdroid-apps commented:
Reviewed.

I believe that you should change the icon from normal paste and also change the palette/tooltip so that the buttons are not identical.

Please don't use Ctrl-B as a shortcut, as that is already bold.

I was unable to use the button to do anything when testing (although I was unable to use the normal paste button).  There was nothing of interest in the logs.  This may be due to me running unstable fedora rawhide.

erilyth commented:
Changed the shortcut to Ctrl-F since nothing else was using it, if you have any other preferred shortcut, Ill change it to that.

Added an icon and set its tooltip as specified.

godiard commented:
I don't understand what is the objective of this patch. What is different when you "Paste special" instead of "Paste"? In what cases? Other than that, the code duplicate all the image management from __paste_button_cb() and that is a bad idea. If needed, move the code to a separated method and call from the two callback methods

erilyth commented:
The paste special pastes text without formatting. The 'Paste Unformatted' option on http://www.abisource.com/help/en-US/interface/menucontextual.html is what uses the paste special call to paste. I realized that this works for images as well, so no need to use the code from paste itself for images.

erilyth commented:
Removed unnecessary image management code .

godiard commented:
@erilyth , this not reply my question. When should a kid use this button? What is the use case? What bug or limitation solve? I try to keep the activity so simple as possible, that means do not add options if are not really needed (if not we end with a full text editor like abiword or libreoffice, and that already exist)

erilyth commented:
@godiard This was a gci task that I saw when I was getting introduced to sugarlabs, I was adviced to do this to get started with sugarlabs 
https://codein.withgoogle.com/tasks/5455686636929024/?sp-status=2&sp-is_exhausted=False&sp-page=1

godiard commented:
@erilyth, ok gsoc approal is on @samdroid-apps hands, even if is not included in the activity

erilyth commented:
@godiard, I wasn't doing this for gci though, just to get familiar with sugar. So if its not needed for the activity, then lets close the PR.
